src/main/java/com/dddkurlyclone/product/application/ProductService.java

```diff
@@ -0,0 +1,27 @@
+package com.dddkurlyclone.product.application;
+
+import com.dddkurlyclone.product.domain.Product;
+import com.dddkurlyclone.product.errors.ProductNotFoundException;
+import com.dddkurlyclone.product.repositories.ProductRepository;
+import java.util.List;
+import javax.transaction.Transactional;
+import org.springframework.stereotype.Service;
+
+/** 상품의 정보를 담고 있습니다. */
+@Service
+@Transactional
+public class ProductService {
+  private final ProductRepository productRepository;
+
+  public ProductService(ProductRepository productRepository) {
+    this.productRepository = productRepository;
+  }
+
+  public List<Product> getProducts() {
+    return productRepository.find();
+  }
+
+  public Product getProduct(Long id) {
+    return productRepository.withId(id).orElseThrow(() -> new ProductNotFoundException(id));
```

Comment:
```suggestion
    return productRepository
      .withId(id)
      .orElseThrow(() -> new ProductNotFoundException(id));
```

이쪽이 가독성이 더 좋지 않을까 제안드립니다.